src/python/pants/engine/target_test.py

```diff
@@ -87,75 +92,50 @@ def test_invalid_fields_rejected() -> None:
     assert "Unrecognized field `invalid_field=True`" in str(exc)
 
 
-def test_get_primitive_field() -> None:
+def test_get_field() -> None:
     extensions = ("GhcExistentialQuantification",)
-    assert (
-        HaskellTarget({HaskellGhcExtensions.alias: extensions}, address=Address.parse(":lib"))
-        .get(HaskellGhcExtensions)
-        .value
-        == extensions
-    )
-
-    # Default value
-    assert (
-        HaskellTarget({}, address=Address.parse(":default")).get(HaskellGhcExtensions).value == ()
-    )
+    tgt = HaskellTarget({HaskellGhcExtensions.alias: extensions}, address=Address.parse(":lib"))
 
+    assert tgt[HaskellGhcExtensions].value == extensions
+    assert tgt.get(HaskellGhcExtensions).value == extensions
+    assert tgt.get(HaskellGhcExtensions, default_raw_value=["GhcDefault"]).value == extensions
 
-def test_get_async_field() -> None:
```

Comment:
It's probably easier to view the diff as separate/not unified. I moved this method lower in the file to reflect that `AsyncField`s are an advanced case and to have the tests for the more common idioms like `tgt.has_field()` and `tgt.get()` be closer together.